File: cordapp-example/workflows-java/src/test/java/com/example/test/NodeDriver.java
package com.example.test;

import com.google.common.collect.ImmutableList;
import com.google.common.collect.ImmutableSet;
import net.corda.core.concurrent.CordaFuture;
import net.corda.core.identity.CordaX500Name;
import net.corda.testing.driver.DriverParameters;
import net.corda.testing.driver.NodeHandle;
import net.corda.testing.driver.NodeParameters;
import net.corda.testing.node.User;

import java.util.List;

import static net.corda.testing.driver.Driver.driver;

/**
 * This file is exclusively for being able to run your nodes through an IDE.
 * Do not use in a production environment.
 */
public class NodeDriver {
    public static void main(String[] args) {
        final User user = new User("user1", "test", ImmutableSet.of("ALL"));
        driver(new DriverParameters().withWaitForAllNodesToFinish(true), dsl -> {
                    List<CordaFuture<NodeHandle>> nodeFutures = ImmutableList.of(
                            dsl.startNode(new NodeParameters()
                                    .withProvidedName(new CordaX500Name("PartyA", "London", "GB"))
                                    .withRpcUsers(ImmutableList.of(user))),
                            dsl.startNode(new NodeParameters()
                                    .withProvidedName(new CordaX500Name("PartyB", "New York", "US"))
                                    .withRpcUsers(ImmutableList.of(user))),
                            dsl.startNode(new NodeParameters()
                                    .withProvidedName(new CordaX500Name("PartyC", "Paris", "FR"))
                                    .withRpcUsers(ImmutableList.of(user))));

                    try {
                        nodeFutures.get(0).get();
                        nodeFutures.get(1).get();
                        nodeFutures.get(2).get();

                    } catch (Throwable e) {
                        System.err.println("Encountered exception in node startup: " + e.getMessage());
                        e.printStackTrace();
                    }

                    return null;
                }
        );
    }
}

File: cordapp-example/workflows-kotlin/src/test/kotlin/com/example/test/NodeDriver.kt
package com.example.test

import net.corda.core.identity.CordaX500Name
import net.corda.core.utilities.getOrThrow
import net.corda.testing.driver.DriverParameters
import net.corda.testing.driver.driver
import net.corda.testing.node.TestCordapp
import net.corda.testing.node.User

/**
 * This file is exclusively for being able to run your nodes through an IDE.
 * Do not use in a production environment.
 */
fun main(args: Array<String>) {
    val user = User("user1", "test", permissions = setOf("ALL"))
    driver(DriverParameters(waitForAllNodesToFinish = true, cordappsForAllNodes = listOf(TestCordapp.findCordapp("com.example.flow")))) {
        val nodeFutures = listOf(
                startNode(
                        providedName = CordaX500Name("PartyA", "London", "GB"),
                        rpcUsers = listOf(user)),
                startNode(
                        providedName = CordaX500Name("PartyB", "New York", "US"),
                        rpcUsers = listOf(user)),
                startNode(
                        providedName = CordaX500Name("PartyC", "Paris", "FR"),
                        rpcUsers = listOf(user)))

        nodeFutures.map { it.getOrThrow() }
    }
}
